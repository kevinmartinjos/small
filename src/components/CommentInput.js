import React from 'react';
import {Button} from 'react-bootstrap';
import FakeServer from '../FakeServer/FakeServer';

var PropTypes = React.PropTypes;

var CommentInput = React.createClass({
	propTypes: {
		submitCallback: PropTypes.func.isRequired,
		postId: PropTypes.string.isRequired
	},
	getInitialState(){
		return {
			value: ''
		};
	},
	getDefaultState(){
		return {
			value: ''
		};
	},
	handleSubmit(event){
		var server = new FakeServer();
		server.addPostComment(this.props.postId, this.state.value);
		this.props.submitCallback();
		this.setState(this.getDefaultState());
		event.preventDefault();
	},
	handleChange(event){
		this.setState({value: event.target.value});
	},
	render() {
		return(
			<div className="CommentInput">
				<form onSubmit={this.handleSubmit}>
					<div>
					<textarea value={this.state.value}
						onChange={this.handleChange}
					/>
					</div>
					<Button bsSize='sm' type='submit' bsStyle='primary'>Submit</Button>
				</form>
			</div>
		);
	}
});

export default CommentInput;
