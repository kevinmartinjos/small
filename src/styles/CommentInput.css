.CommentInput{
	padding-top: 10px;
	padding-bottom: 10px;
}

.CommentInput button{
	margin-top: 10px;
	margin-bottom: 10px;
}