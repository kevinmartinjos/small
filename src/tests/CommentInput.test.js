import React from 'react';
import renderer from 'react-test-renderer';
import CommentInput from '../components/CommentInput';
import ReactTestUtils from 'react-addons-test-utils';

test('CommentInput should be rendered', () => {
	const component = renderer.create(
		<CommentInput submitCallback={function(){}} postId="1"/>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Should reload comments on submit', () => {
	var flag = 0;

	//onSubmit is passed as a callback. onSubmit *could* reload all comments of a post
	//when a new comment is submitted
	var onSubmit = function(){
		flag = 1;
	};

	//mocking an event object to pass to handleSubmit
	var event = {};
	event.preventDefault = function(){};

	let component = ReactTestUtils.renderIntoDocument(<CommentInput submitCallback={onSubmit} postId="1"/>);
  	component.handleSubmit(event);
  	
  	expect(flag).toEqual(1);
});