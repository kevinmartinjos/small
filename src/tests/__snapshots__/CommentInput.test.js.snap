exports[`test CommentInput should be rendered 1`] = `
<div
  className="CommentInput">
  <form
    onSubmit={[Function]}>
    <div>
      <textarea
        onChange={[Function]}
        value="" />
    </div>
    <button
      className="btn btn-sm btn-primary"
      disabled={false}
      type="submit">
      Submit
    </button>
  </form>
</div>
`;
